pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -634,17 +637,21 @@ func (c *Cluster) updateMon(m *monConfig, d *apps.Deployment) error {
 
 	daemonType := string(config.MonType)
 	var cephVersionToUse cephver.CephVersion
-	currentCephVersion, err := client.LeastUptodateDaemonVersion(c.context, c.ClusterInfo.Name, daemonType)
-	if err != nil {
-		logger.Warningf("failed to retrieve current ceph %s version. %+v", daemonType, err)
-		logger.Debug("could not detect ceph version during update, this is likely an initial bootstrap, proceeding with c.clusterInfo.CephVersion")
-		cephVersionToUse = c.ClusterInfo.CephVersion
-	} else {
-		logger.Debugf("current cluster version for monitors before upgrading is: %+v", currentCephVersion)
-		cephVersionToUse = currentCephVersion
+
+	// If this is not an upgrade there is no need to check the ceph version
+	if c.isUpgrade {
+		currentCephVersion, err := client.LeastUptodateDaemonVersion(c.context, c.ClusterInfo.Name, daemonType)
+		if err != nil {
+			logger.Warningf("failed to retrieve current ceph %s version. %+v", daemonType, err)
+			logger.Debug("could not detect ceph version during update, this is likely an initial bootstrap, proceeding with c.clusterInfo.CephVersion")
+			cephVersionToUse = c.ClusterInfo.CephVersion
+		} else {
+			logger.Debugf("current cluster version for monitors before upgrading is: %+v", currentCephVersion)
+			cephVersionToUse = currentCephVersion
+		}
 	}
 
-	err = updateDeploymentAndWait(c.context, d, c.Namespace, daemonType, m.DaemonName, cephVersionToUse)
+	err := updateDeploymentAndWait(c.context, d, c.Namespace, daemonType, m.DaemonName, cephVersionToUse, c.isUpgrade)
```

Comment:
If you have 3 mons, 1 is down AND you're changing the spec image which leads to different Ceph version, then an upgrade will be triggered (`c.isUpgrade ` will be true), then check `ok-to-stop` will run. At this point, if a mon is down, the orchestration will wait until the dead mon is rescheduled/healthy, things will be blocked until then.

What's you've experienced in the past won't happen again because `ok-to-stop` checks will be skipped.
Does that answer your concern?